Pt 2 create parser as a rake task

SergeiKaptelin opened the pull request:
- create parser as a rake task
- add validations to lesson

SergeiKaptelin committed:
add nokogiri

SergeiKaptelin committed:
add rake task

update validations for faculty and group models;
parse faculty, group and save in models.

SergeiKaptelin committed:
make type and short_name in group separately

make click_information clean

SergeiKaptelin committed:
improve lessons information from array to hash for easy access

SergeiKaptelin committed:
change time_zone to Kiev, change type column to lesson_type in lessons model

SergeiKaptelin committed:
refactoring code to ease read

SergeiKaptelin committed:
add rebuild_timetable to ease work with parse_lessons

SergeiKaptelin committed:
add save lesson in db

SergeiKaptelin committed:
improve dut_parser with handle specific changes in timetable

SergeiKaptelin committed:
move compare_parse_lessons_and_db_lessons to Lesson model

move delete_empty_dates to Lesson model

SergeiKaptelin committed:
make clean code for parse all lessons

SergeiKaptelin committed:
move compare lessons and parsed lessons to dut_parser

SergeiKaptelin committed:
Merge remote-tracking branch 'origin/master' into PT-2-create-parser-as-a-rake-task

# Conflicts:
#	Gemfile.lock

SergeiKaptelin committed:
add parsing through time intervals

SergeiKaptelin committed:
improve parser to correct handle changes in timetable

LinchSmyth reviewed (changes requested):
View slack chat for details

SergeiKaptelin committed:
make code more readable

- delete custom find_by_or_create
- update time_intervals to ease understanding

Commit:
final changes to dut_parser.rake & Lesson before merging
